cpp/src/arrow/buffer_builder.h

```diff
@@ -198,6 +203,7 @@ class ARROW_EXPORT BufferBuilder {
   uint8_t* data_;
   int64_t capacity_;
   int64_t size_;
+  int64_t alignment_;
```

Comment:
```suggestion
  const int64_t alignment_;
```

Reply:
This seems to cause a bunch of compilation issues for me: 

```
/Users/sasha/arrow/cpp/src/arrow/array/builder_base.h:76:3: error: explicitly defaulted move assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]

...

/Users/sasha/arrow/cpp/src/arrow/buffer_builder.h:206:17: note: copy assignment operator of 'BufferBuilder' is implicitly deleted because field 'alignment_' is of const-qualified type 'const int64_t' (aka 'const long long')
  const int64_t alignment_;
```

Reply:
It took me a minute to think this one through, default move assignment isn't allowed because it would boil down to...

```
BufferBuilder& BufferBuilder(BufferBuilder&& other) {
  // ...
  // Not allowed
  this->alignment_ = std::move(other.alignment_);
  // ...
}
```